userspace/falco/falco.cpp

```diff
@@ -1472,66 +1095,67 @@ int falco_init(int argc, char **argv)
 		//
 		// Run k8s, if required
 		//
-		if(k8s_api)
+		if(!app.options().k8s_api.empty())
 		{
-			if(!k8s_api_cert)
+			// Create string pointers for some config vars
+			// and pass to inspector. The inspector then
+			// owns the pointers.
+			std::string *k8s_api_ptr = new string(app.options().k8s_api);
+			std::string *k8s_api_cert_ptr = new string(app.options().k8s_api_cert);
+			std::string *k8s_node_name_ptr = new string(app.options().k8s_node_name);
+
+			if(k8s_api_cert_ptr->empty())
 			{
 				if(char* k8s_cert_env = getenv("FALCO_K8S_API_CERT"))
 				{
-					k8s_api_cert = new string(k8s_cert_env);
+					*k8s_api_cert_ptr = k8s_cert_env;
 				}
 			}
-			inspector->init_k8s_client(k8s_api, k8s_api_cert, k8s_node_name, verbose);
-			k8s_api = 0;
-			k8s_api_cert = 0;
+			inspector->init_k8s_client(k8s_api_ptr, k8s_api_cert_ptr, k8s_node_name_ptr, app.options().verbose);
 		}
 		else if(char* k8s_api_env = getenv("FALCO_K8S_API"))
 		{
-			if(k8s_api_env != NULL)
+			// Create string pointers for some config vars
```

Comment:
At first glance, this looks like the exact same [block at line 1100](https://github.com/falcosecurity/falco/pull/1886/files#diff-67ae49cc953e68396a677916d8dce080ef900deede8f39489ae611dce4b3cab2R1100). If that's so, it would be great to handle the if-else `k8s_api` string init logic before, and then have a single block for initializing the k8s client.

Reply:
Sure, the old blocks were nearly identical too, but I can reorganize this to avoid the duplicate code.